### PR TITLE
LXD: update to 3.19, make sqlite-replication not conflict with sqlite

### LINKED ISSUE #18310: [Bug] sqlite-replication replacing sqlite.
As noticed in https://github.com/void-linux/void-packages/issues/18179, sqlite-replication which is the dependency of exactly one package (dqlite) replaces the system sqlite and therefor sqlite for all installed programs.
There is really no point in providing a patched version of sqlite just for this one package, it should either statically link the patched sqlite or use a different soname.

@CameronNemo

### PR DESCRIPTION
**Pull request by CameronNemo**
Fixes #18310 (but no commit blurb, whoops)

### CODE REVIEW

File: common/shlibs [hunk 1]
```diff
@@ -3705,3 +3705,6 @@ libnvpair.so.1 zfs-0.8.2_1
 libjsonnet.so.0 jsonnet-0.14.0_2
 libjsonnet++.so.0 jsonnet-0.14.0_2
 libigdgmm.so.11 intel-gmmlib-19.4.1_1
+libigdgmm.so.12 intel-gmmlib-19.3.4_1
```
Comment (xtraeme) [inline common/shlibs]: why did you add this unrelated change?

File: srcpkgs/dqlite/template [hunk 2]
```diff
@@ -1,24 +1,31 @@
...
-license="Apache-2.0"
-homepage="https://github.com/CanonicalLtd/dqlite"
+license="custom:LGPL-3.0-only-linking-exception"
+homepage="https://github.com/canonical/dqlite"
 distfiles="${homepage}/archive/v${version}.tar.gz"
-checksum=47d31550d970a12d25101c331896cfb855128c110d912f5da3371062d9ece52b
+checksum=22e3d281c396cc409e4afced20e143c4a014b036146945e30f9ebf3bee410865
+
+_libdir="${XBPS_CROSS_BASE}/usr/lib/sqlite-replication"
+export LDFLAGS+=" -L${_libdir} -Wl,-R${_libdir}"
```
Comment (xtraeme) [inline srcpkgs/dqlite/template]: You can get rid of `export` and simply set it like:

```
LDFLAGS="-L${_libdir} -Wl,-R${_libdir}"
```

File: srcpkgs/lxd/template [hunk 3]
```diff
@@ -1,37 +1,27 @@
...
-}
-
-do_build() {
-	# don't go-get the dependencies, just install with what's there
-	cd "$GOSRCPATH"
-	go_package=${go_package:-$go_import_path}
-	go install -tags "${go_build_tags}" -ldflags "${go_ldflags}" ${go_package}
-}
+_libdir="${XBPS_CROSS_BASE}/usr/lib/sqlite-replication"
+export LDFLAGS+=" -L${_libdir} -Wl,-R${_libdir}"
```
Comment (xtraeme) [inline srcpkgs/lxd/template]: ditto

File: srcpkgs/raft/template [hunk 4]
```diff
@@ -0,0 +1,33 @@
+# Template file for 'raft'
+pkgname=raft
+version=0.9.16
+revision=1
+build_style=gnu-configure
+configure_args="--enable-example=no"
+hostmakedepends="pkg-config autoconf automake libtool"
```
Comment (xtraeme) [inline srcpkgs/raft/template]: `autoconf` is redundant, anyway it's harmless

File: common/shlibs (hunk 1 above)
Comment (CameronNemo) [reply common/shlibs]: No idea. Was an artifact of a rebase or something.

File: srcpkgs/sqlite-replication/template [hunk 5]
```diff
@@ -29,15 +28,21 @@ pre_configure() {
 	sed -i -e 's/ -ltinfo//g' configure
 }
 
+post_install() {
+	vmkdir usr/lib/pkgconfig
+	mv "$DESTDIR"/usr/lib{/sqlite-replication,}/pkgconfig/sqlite3.pc
+	# Alpine just deletes the binary, but someone may want it for debugging
+	mv "$DESTDIR"/usr/bin/sqlite3{,.sqlite-replication}
```
Comment (fosslinux) [inline srcpkgs/sqlite-replication/template]: Imo this should be `vmove`d to a separate package.

File: srcpkgs/sqlite-replication/template [hunk 6]
```diff
@@ -29,15 +28,21 @@ pre_configure() {
...
+}
+
 sqlite-replication-devel_package() {
 	depends="sqlite-replication>=${version}_${revision}"
-	replaces="sqlite-devel>=0"
-	provides="sqlite-devel-${version}_${revision}"
+	conflicts="sqlite-devel>=0"
```
Comment (fosslinux) [inline srcpkgs/sqlite-replication/template]: Why does this conflict? Isn't the point of this PR to make sqlite-replication not conflict with sqlite? Or is -devel exempt?

File: srcpkgs/sqlite-replication/template [hunk 7]
```diff
@@ -29,15 +28,21 @@ pre_configure() {
...
 	short_desc+=" - development files"
 	pkg_install() {
 		vmove usr/include
```
Comment (fosslinux) [inline srcpkgs/sqlite-replication/template]: If we want to make -devel not conflict the header files probably need to change.

File: srcpkgs/sqlite-replication/template [hunk 8]
```diff
@@ -29,15 +28,21 @@ pre_configure() {
...
 		vmove usr/lib/pkgconfig
```
Comment (fosslinux) [inline srcpkgs/sqlite-replication/template]: Same as above.

File: srcpkgs/lxd/template [hunk 9]
```diff
@@ -1,37 +1,29 @@
...
-	go install -tags "${go_build_tags}" -ldflags "${go_ldflags}" ${go_package}
-}
+_libdir="${XBPS_CROSS_BASE}/usr/lib/sqlite-replication"
+LDFLAGS="-L${_libdir} -Wl,-R${_libdir}"
+# whitelist libcap LDFLAGS (see: https://github.com/lxc/lxd/issues/6727)
+export CGO_LDFLAGS_ALLOW='-Wl,-wrap,pthread_create'
 
 post_install() {
 	vinstall scripts/bash/lxd-client 644 /usr/share/bash-completion/completions lxd
-	vlicense COPYING LICENSE
```
Comment (fosslinux) [inline srcpkgs/lxd/template]: Why remove this, shouldn't Apache 2.0 be vlicensed?

File: srcpkgs/sqlite-replication/template (hunk 6 above)
Comment (CameronNemo) [reply srcpkgs/sqlite-replication/template]: Cannot link to both at once, so I do not care about the -devel packages conflicting.

File: srcpkgs/sqlite-replication/template (hunk 7 above)
Comment (CameronNemo) [reply srcpkgs/sqlite-replication/template]: the directory would need to change, but that is just extra work for no gain

File: srcpkgs/lxd/template (hunk 9 above)
Comment (CameronNemo) [reply srcpkgs/lxd/template]: nope it should not be

File: srcpkgs/sqlite-replication/template (hunk 5 above)
Comment (CameronNemo) [reply srcpkgs/sqlite-replication/template]: why though?

Comment (fosslinux) [reply srcpkgs/sqlite-replication/template]: It's unnecessary for normal use of the package, and you say keep it for debugging; hence why I personally think it should be in a separate package.

Comment (CameronNemo) [reply srcpkgs/sqlite-replication/template]: made the change

File: srcpkgs/raft/template (hunk 4 above)
Comment (CameronNemo) [reply srcpkgs/raft/template]: fixed it for one template, forgot for the other. not worth the CPU for another CI round IMO. we have a planet to save, after all.

**Comment by CameronNemo** [comment]
@fosslinux I updated the template in a different way to handle lib32

@Duncaen I assume you are interested in the sqlite-replication portion of this PR

**Comment by fosslinux** [comment]
@CameronNemo I get `lxd: symbol lookup error: /usr/lib/libdqlite.so.0: undefined symbol: sqlite3_wal_replication_unregister` trying to run this on my pi.

**Comment by fosslinux** [comment]
Ah, I think I see the problem. In the masterdir it links against `/usr/armv7l-linux-gnueabihf/usr/lib/sqlite-replication`, however in the resulting system it's need to be linked to `/usr/lib/sqlite-replication`. For armv7l for example. I'm unsure how to fix this...

**Comment by fosslinux** [comment]
Actually, maybe not.

**Comment by CameronNemo** [comment]
I addressed that in the most recent push. Thanks for pointing it out.

**Comment by fosslinux** [comment]
hoorayy~~!

**Comment by ernierasta** [comment]
Hi, first, for great work, but I still have the problem, up-to-date void x86_64.

``` bash
lxd: symbol lookup error: /usr/lib/libdqlite.so.0: undefined symbol: sqlite3_wal_replication_unregister
```
**dqlite-1.3.1_1**
**sqlite-replication-3.30.1_2**
**lxd-3.14_1**

**Comment by fosslinux** [comment]
Hi,

Your issue is that you have mismatched versions of packages, AFAICT. The lxd version you should be using is 3.19.

**Comment by ernierasta** [comment]
Thank you @fosslinux! You are right. Probably we are waiting for workers to finish building or my mirror is a bit late. I will wait longer.

**Comment by CameronNemo** [comment]
@ernierasta LXD 3.19 has been built for a while. Does it not prompt you to upgrade with a normal update? What do the following command tell you?

* `xbps-query -R lxd`
* `xbps-query lxd`
* `xbps-query -H`

**Comment by CameronNemo** [comment]
Actually that is my bad... looks like the build failed. https://build.voidlinux.org/builders/x86_64_builder/builds/22710/steps/shell_3/logs/stdio

**Comment by CameronNemo** [comment]
Fixed, it seems: https://github.com/void-linux/void-packages/pull/18750
